src/scheduler/mod.rs

```diff
@@ -689,91 +695,99 @@ mod test {
             .root
             .clone();
 
+        let result_state = Arc::new(Mutex::new(Vec::<usize>::new()));
         let num_resources = 3;
         let resources = (0..num_resources).map(|id| Rsrc { id }).collect::<Vec<_>>();
 
         let mut expected = Vec::new();
 
-        let scheduler_handle = Scheduler::start(scheduler).expect("Failed to start scheduler.");
-
+        let mut futures = Vec::new();
         for i in 0..num_resources {
+            let result_state = Arc::clone(&result_state);
             // Schedule a slow task to tie up all the resources
             // while the next batch of `Task1`s is scheduled.
             let priority = 0;
             expected.push(0);
-            scheduler.lock().unwrap().schedule(
+            futures.push(scheduler.lock().unwrap().schedule(
```

Comment:
`schedule(…)` returns an ordinary `Receiver`, not a future — so let's not call these results `futures`. It's confusing.

Reply:
hmm, right, how about `chans`?

Reply:
I'm fine with that.

Reply:
Done